packages/webpack/mixins/esbuild/importComponentTransform.js

```diff
@@ -0,0 +1,10 @@
+const regex = /importComponent\s*\(\s*\(\)\s+=>\s+import\(\s*'([^']+)'\s*\)\s*\)/g;
+
+function transformImportComponent(source) {
+  return source.replace(
+    regex,
+    "importComponent({ load: () => import('$1'), moduleId: require.resolveWeak('$1') })"
+  );
+}
+
+module.exports = transformImportComponent;
```

Comment:
I think this should be named something with `loader` instead of `transform`. Otherwise I find it pretty confusing.

Reply:
On the other hand, it isn't loading anything but it is actually transforming a string 🤷‍♂️

Reply:
I didn't tell Webpack to call their transorms `loader`. 😅